data_model/src/lib.rs

```diff
@@ -368,6 +382,70 @@ pub mod parameter {
         }
     }
 
+    /// Convenience tool for setting parameters
+    #[derive(Default)]
+    pub struct ParametersBuilder {
+        parameters: Vec<Parameter>,
+    }
+
+    /// Error associated with parameters builder
+    #[derive(From, Debug, Display, Copy, Clone)]
+    pub enum ParametersBuilderError {
+        /// Error emerged during parsing of parameter id
+        Parse(ParseError),
+    }
+
+    #[cfg(feature = "std")]
+    impl std::error::Error for ParametersBuilderError {}
+
+    impl ParametersBuilder {
+        /// Construct [`Self`]
+        pub fn new() -> Self {
+            Self::default()
+        }
+
+        /// Add [`Parameter`] to self
+        ///
+        /// # Errors
+        /// - [`ParameterId`] parsing failed
+        pub fn parameter(
+            mut self,
+            parameter_id: &str,
+            val: impl Into<Value>,
+        ) -> Result<Self, ParametersBuilderError> {
+            let parameter = Parameter {
+                id: parameter_id.parse()?,
+                val: val.into(),
+            };
+            self.parameters.push(parameter);
+            Ok(self)
+        }
+
+        /// Create sequence isi for setting parameters
+        pub fn into_set_parameters(self) -> isi::SequenceBox {
+            isi::SequenceBox {
+                instructions: self
+                    .parameters
+                    .into_iter()
+                    .map(isi::SetParameterBox::new)
+                    .map(Into::into)
+                    .collect(),
+            }
+        }
+
+        /// Create sequence isi for creating parameters
+        pub fn into_create_parameters(self) -> isi::SequenceBox {
+            isi::SequenceBox {
+                instructions: self
+                    .parameters
+                    .into_iter()
+                    .map(isi::NewParameterBox::new)
```

Comment:
I don't know where exactly to comment, but I see that we have 
```rs
pub struct NewParameter<P: Identifiable> {
    pub parameter: P,
}
```

but I think we should have:
```rs
pub struct NewParameter {
    pub id: ParameterId,
    #[getset(get = "pub")]
    pub val: Value,
}
```

the same goes for `SetParameter`. There is no point in making them parametrized

Reply:
you can also do:
```rs
pub struct NewParameter(Parameter);
```

I prefer this actually

Reply:
I want to do least amount of change but remove this generic, so i think like this should be fine

```rust
pub struct NewParameter {
    parameter: Parameter
}
```